File: Lesson 2/CyclicRotation/CyclicRotation.cs
using System;
using System.Linq;
// you can also use other imports, for example:
// using System.Collections.Generic;

// you can write to stdout for debugging purposes, e.g.
// Console.WriteLine("this is a debug message");

class Solution {	
	
    public int [] solution(int[] A, int K)
	{    
		return A.Select((x, i) => A[(A.Length - K % A.Length + i) % A.Length]).ToArray();
    }	
	

	public int[] solution(int[] A, int K) {
        // write your code in C# 6.0 with .NET 4.5 (Mono)
        int len=A.Length;
        if (len==0)
            return new int[] { };    
            
        int Count = K % len;
        int[] B = new int[len];
        for (int i = 0; i < len; i++)
        {
            B[(i + Count) % len] = A[i];
        }
        return B;

    }
}

File: Lesson 2/CyclicRotation/CyclicRotation.py
def solution(A, K): 
	if not A:
        return []
    
	K = K % len(A)
    m = len(A)
	return  A[m - K:]+A[:m - K] 

	
	
def solution(A, K):
    # write your code in Python 3.6
    if not A:
        return A
        
    K = K % len(A)
    
    for i in range(K):
        A = A[-1:] + A[:-1]
    return A	

	
